src/object.rs

```diff
@@ -655,6 +655,9 @@ pub trait ObjectExt: IsA<Object> {
     ///
     /// [downgrade_sync]: trait.SyncObjectExt.html#method.downgrade_sync
     fn downgrade(&self) -> WeakRef<Self>;
+
+    fn bind_property<'a, 'b, O: IsA<Object>, N: Into<&'a str>, M: Into<&'a str>>(&self, source_property: N, target: &O, target_property: M, flags: ::BindingFlags) -> Option<::Binding>;
+    fn bind_property_full<'a, 'b, O: IsA<Object>, N: Into<&'a str>, M: Into<&'a str>, F: Fn(&::Binding, &Value) -> Option<Value> + Send + Sync + 'static, G: Fn(&::Binding, &Value) -> Option<Value> + Send + Sync + 'static>(&self, source_property: N, target: &O, target_property: M, flags: ::BindingFlags, transform_to: Option<F>, transform_from: Option<G>) -> Option<::Binding>;
```

Comment:
This is suboptimal as you can't just pass `None`, it needs type annotations everywhere.

Opinions: This either needs 3 functions (first, second, both closures) or a builder (then all functions can be combined). IMHO a builder is nicer, then the flags could also become optional for example.

Reply:
Builder it is if nobody complains :)